src/postgres/abstract/AbstractPostgresClient.ts

```diff
@@ -0,0 +1,83 @@
+        this.postgresFlexibleClient = new PostgreSQLFlexibleManagementClient(credentials, subscriptionId, options);
+        this.postgresSingleClient = new PostgreSQLManagementClient(credentials, subscriptionId, options);
+    }
+
+    async checkNameAvailability(serverType: Models.PostgresServerType, name: string) : Promise<Models.AbstractNameAvailability> {
+        switch (serverType){
+            case Models.PostgresServerType.Flexible:
+                return this.postgresFlexibleClient.checkNameAvailability.execute({name: name, type: "Microsoft.DBforPostgreSQL"});
+            case Models.PostgresServerType.Single:
+                return this.postgresSingleClient.checkNameAvailability.execute({name: name, type: "Microsoft.DBforPostgreSQL"});
+            default:
+                throw new Error("Service not implemented.");
+        }
+    }
+
+    async createFirewallRule(serverType: Models.PostgresServerType, resourceGroup: string, name: string, ruleName: string, rule: Models.AbstractFirewallRule): Promise<Models.AbstractFirewallRule> {
+        switch (serverType){
+            case Models.PostgresServerType.Flexible:
+                return this.postgresFlexibleClient.firewallRules.createOrUpdate(resourceGroup, name, ruleName, rule);
+            case Models.PostgresServerType.Single:
+                return this.postgresSingleClient.firewallRules.createOrUpdate(resourceGroup, name, ruleName, rule);
+            default:
+                throw new Error("Service not implemented.");
+        }
+    }
+
+    async createServer(serverType: Models.PostgresServerType, resourceGroup: string, name: string, parameters: Models.AbstractServerCreate): Promise<Models.PostgresAbstractServer> {
+        switch (serverType){
+            case Models.PostgresServerType.Flexible:
+                return flexibleAsAbstractServer(await this.postgresFlexibleClient.servers.create(resourceGroup, name, asFlexibleParameters(parameters)));
+            case Models.PostgresServerType.Single:
+                return singleAsAbstractServer(await this.postgresSingleClient.servers.create(resourceGroup, name, asSingleParameters(parameters)));
+            default:
+                throw new Error("Service not implemented.");
+        }
+    }
+
+    async deleteServer(serverType: Models.PostgresServerType, resourceGroup: string, name: string): Promise<msRest.RestResponse> {
+        switch (serverType){
+            case Models.PostgresServerType.Flexible:
+                return this.postgresFlexibleClient.servers.deleteMethod(resourceGroup, name);
+            case Models.PostgresServerType.Single:
+                return this.postgresSingleClient.servers.deleteMethod(resourceGroup, name);
+            default:
+                throw new Error("Service not implemented.");
+        }
+    }
+
+    async listServers(): Promise<Models.PostgresAbstractServerList> {
```

Comment:
```suggestion
    async listAllServers(): Promise<Models.PostgresAbstractServerList> {
```

It doesn't have to be this exact name but I want to convey that it's listing flexible and single servers. And it's the only function in this class that doesn't just do one or the other